### DISCUSSION TITLE
Blocked Gibbs Sampling Generation

### DISCUSSION THREAD

**Discussion by martinez-zacharya** [facebookresearch/esm#462] [General]
Is there a script or colab notebook for the gibbs sampling protein generation method from this paper? https://www.biorxiv.org/content/10.1101/2022.12.21.521521v1.full#F1

**Comment by tony-res** [comment]
[TRILL](https://trill.readthedocs.io/en/latest/home.html) has Gibbs sampling implemented.

> ESM2 Gibbs: Using Gibbs sampling, you can generate synthetic proteins from a finetuned ESM2 model
> $ trill example_4 1 generate ESM2_Gibbs --finetuned /path/to/finetuned_model.pt --esm2_arch esm2_t30_150M_UR

**Reply by tony-res** [reply]
Oh. It's Zachary asking. Well, I guess you knew that answer, then!
